### PR TITLE
test(clusterconfig): missing unit tests for cluster infrastructure gatherer

### PR DESCRIPTION
**Pull request by ncaak**
This PR implements new unit tests for infraestructure gatherer

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [X] Others (CI, Infrastructure, Documentation)

## Sample Archive

- N/A

## Documentation

- N/A

## Unit Tests

- `pkg/gatherers/clusterconfig/gather_cluster_infrastructure_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Changelog

Pending

## Breaking Changes

No

## References

https://issues.redhat.com/browse/CCXDEV-10105

### CODE REVIEW

File: pkg/gatherers/clusterconfig/gather_cluster_infrastructure_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+						Resource: &v1.Infrastructure{
+							ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+							Status:     v1.InfrastructureStatus{PlatformStatus: &v1.PlatformStatus{}},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:   "Retrieving no infraestructure returns no error/no record",
```
[the remark this answers is not in the document]
Comment (ncaak) [reply pkg/gatherers/clusterconfig/gather_cluster_infrastructure_test.go:40]: ahhh :( Sorry, this comes from Spanish-English translation. I already fixed it on other instances but this one was missed.

**Comment by tremes** [comment]
Thanks. Overriding since it's a new unit test
/lgtm 
/override ci/prow/e2e-metal-ipi-ovn-ipv6
/override ci/prow/e2e-agnostic-upgrade 
/override ci/prow/e2e

**Comment by tremes** [comment]
Just a unit test. No need to test upgrade.
/override ci/prow/e2e-agnostic-upgrade

**Comment by tremes** [comment]
JUst a unit test.
/override ci/prow/e2e

**Pull request merged**